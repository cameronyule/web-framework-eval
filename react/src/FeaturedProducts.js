import React from 'react';

import FeaturedProductsList from './FeaturedProductsList';

export default class FeaturedProducts extends React.Component {
  constructor() {
    super();

    this.state = {data: [
      {id: 1, href: "#", title: "Your working week, sorted", linktext: "Shop 9-5 style"},
      {id: 2, href: "#", title: "Half your daily uniform", linktext: "Shop tops"},
      {id: 3, href: "#", title: "Let's dress this bump", linktext: "Shop dresses"},
      {id: 4, href: "#", title: "Layers to lean on", linktext: "Shop maternity essentials"}
    ]};
  }

  render() {
    return (
      <section className="featuredproducts">
        <FeaturedProductsList data={this.state.data} />
      </section>
    )
  }
}
