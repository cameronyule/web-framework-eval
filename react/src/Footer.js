import React from 'react';

const Footer = () => (
  <footer>
    <p>&copy; 2016</p>
  </footer>
);

export default Footer;
