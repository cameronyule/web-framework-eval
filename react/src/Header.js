import React from 'react';

const Header = (props) =>
  (
    <header>
      <h1>{props.title}</h1>
    </header>
  );

Header.propTypes = {
  title: React.PropTypes.string
}

Header.defaultProps = {
  title: "Header"
}

export default Header;
