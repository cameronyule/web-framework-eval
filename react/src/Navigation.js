import React from 'react';

import NavigationLink from './NavigationLink';

export default class Navigation extends React.Component {
  constructor() {
    super();

    this.state = {data: [
      {id: 1, href: "#", title: "Maternity Clothes"},
      {id: 2, href: "#", title: "New Arrivals"},
      {id: 3, href: "#", title: "Dresses"},
      {id: 4, href: "#", title: "Tops"},
      {id: 5, href: "#", title: "Tees"}
    ]};
  }

  render() {
    const navigationNodes = this.state.data.map(function(navitem) {
      return (
        <NavigationLink key={navitem.id} href={navitem.href} title={navitem.title} />
      )
    });

    return (
      <nav>
        <ul>
          {navigationNodes}
        </ul>
      </nav>
    )
  }
}
