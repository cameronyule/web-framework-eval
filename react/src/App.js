import React from 'react';

import Header from './Header';
import Navigation from './Navigation';
import FeaturedProducts from './FeaturedProducts';
import Footer from './Footer';

const navigation = [
  {id: 1, href: '#', title: 'Coats & Jackets'},
  {id: 2, href: '#', title: 'Knitwear'},
  {id: 3, href: '#', title: 'Shirts'},
  {id: 4, href: '#', title: 'Tops'},
  {id: 5, href: '#', title: 'Trousers'}
];

const products = [
  {id: 1, href: '#', title: 'Knit Picks', description: `Versatility, quality
  and diverse design mean that in AW16’s knitwear collection, there’s a style
  to suit every man no matter his taste or need.`},
  {id: 2, href: '#', title: 'Collaborations', description: `A collection of
  products which will enable your life to be lived that little bit more
  stylishly.`},
  {id: 3, href: '#', title: 'The Green Light', description: `Don’t let your
  autumn/winter wardrobe become a uniform of black, navy and grey. For AW16
  green has received the green light.`},
  {id: 4, href: '#', title: 'Coat Tales', description: `Whether you favour a
  short jacket or a long overcoat, this season’s outerwear is a tale of two
  halves. Discover AW16’s most essential investments.`}
];

const App = () =>
  (
    <div>
      <Header />
      <Navigation data={navigation} />
      <FeaturedProducts data={products} />
      <Footer />
    </div>
  );

export default App;
