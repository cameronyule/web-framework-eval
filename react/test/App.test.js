import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import App from '../src/App';
import Header from '../src/Header';
import Navigation from '../src/Navigation';
import FeaturedProducts from '../src/FeaturedProducts';
import Footer from '../src/Footer';

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has a header component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Header).length).toEqual(1);
  });

  it('has a navigation component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Navigation).length).toEqual(1);
  });

  it('has a featured products component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(FeaturedProducts).length).toEqual(1);
  });

  it('has a footer component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Footer).length).toEqual(1);
  });
});
