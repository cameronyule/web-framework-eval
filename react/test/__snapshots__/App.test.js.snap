exports[`App renders correctly 1`] = `
<div>
  <Header
    title="Header" />
  <Navigation
    data={
      Array [
        Object {
          "href": "#",
          "id": 1,
          "title": "Coats & Jackets",
        },
        Object {
          "href": "#",
          "id": 2,
          "title": "Knitwear",
        },
        Object {
          "href": "#",
          "id": 3,
          "title": "Shirts",
        },
        Object {
          "href": "#",
          "id": 4,
          "title": "Tops",
        },
        Object {
          "href": "#",
          "id": 5,
          "title": "Trousers",
        },
      ]
    } />
  <FeaturedProducts
    data={
      Array [
        Object {
          "description": "Versatility, quality
        and diverse design mean that in AW16’s knitwear collection, there’s a style
        to suit every man no matter his taste or need.",
          "href": "#",
          "id": 1,
          "title": "Knit Picks",
        },
        Object {
          "description": "A collection of
        products which will enable your life to be lived that little bit more
        stylishly.",
          "href": "#",
          "id": 2,
          "title": "Collaborations",
        },
        Object {
          "description": "Don’t let your
        autumn/winter wardrobe become a uniform of black, navy and grey. For AW16
        green has received the green light.",
          "href": "#",
          "id": 3,
          "title": "The Green Light",
        },
        Object {
          "description": "Whether you favour a
        short jacket or a long overcoat, this season’s outerwear is a tale of two
        halves. Discover AW16’s most essential investments.",
          "href": "#",
          "id": 4,
          "title": "Coat Tales",
        },
      ]
    } />
  <Footer />
</div>
`;
